src/main/java/org/jabref/gui/importer/fetcher/WebSearchPaneViewModel.java

```diff
@@ -91,13 +104,40 @@ public void search() {
 
         SearchBasedFetcher activeFetcher = getSelectedFetcher();
 
-        BackgroundTask<ParserResult> task = BackgroundTask.wrap(() -> new ParserResult(activeFetcher.performSearch(getQuery().trim())))
-                                                          .withInitialMessage(Localization.lang("Processing %0", getQuery()));
-
+        BackgroundTask<ParserResult> task;
+        QueryParser queryParser = new QueryParser();
+        Optional<ComplexSearchQuery> generatedQuery = queryParser.parseQueryStringIntoComplexQuery(getQuery());
+        if (generatedQuery.isPresent()) {
+            task = BackgroundTask.wrap(() -> new ParserResult(activeFetcher.performComplexSearch(generatedQuery.get())))
+                                 .withInitialMessage(Localization.lang("Processing %0", getQuery()));
+        } else {
+            task = BackgroundTask.wrap(() -> new ParserResult(activeFetcher.performSearch(getQuery().trim())))
+                                 .withInitialMessage(Localization.lang("Processing %0", getQuery()));
+        }
         task.onFailure(dialogService::showErrorDialogAndWait);
 
         ImportEntriesDialog dialog = new ImportEntriesDialog(frame.getCurrentBasePanel().getBibDatabaseContext(), task);
         dialog.setTitle(activeFetcher.getName());
         dialog.showAndWait();
     }
+
+    public void validateQueryStringAndGiveColorFeedback(TextField querySource, String queryString) {
+        Matcher queryValidation = queryPattern.matcher(queryString.strip());
+        if (queryValidation.matches()) {
+            if (containsYearAndYearRange(queryString)) {
+                querySource.setStyle("-fx-background-color: rgba(240, 128, 128, 0.5);");
```

Comment:
Pseudo-classes could be used instead of hardcoding the colors here. See gui/util/ViewModelListCellFactory